azure-kusto-ingest/azure/kusto/ingest/ingest_client.py

```diff
@@ -91,19 +100,22 @@ def ingest_from_blob(self, blob_descriptor: BlobDescriptor, ingestion_properties
         :param azure.kusto.ingest.BlobDescriptor blob_descriptor: An object that contains a description of the blob to be ingested.
         :param azure.kusto.ingest.IngestionProperties ingestion_properties: Ingestion properties.
         """
+        if self._is_closed:
+            raise KustoClosedError()
+
         try:
             queues = self._resource_manager.get_ingestion_queues()
         except KustoServiceError as ex:
             self._validate_endpoint_service_type()
             raise ex
 
         random_queue = random.choice(queues)
-        queue_service = QueueServiceClient(random_queue.account_uri, proxies=self._proxy_dict)
-        authorization_context = self._resource_manager.get_authorization_context()
-        ingestion_blob_info = IngestionBlobInfo(blob_descriptor, ingestion_properties=ingestion_properties, auth_context=authorization_context)
-        ingestion_blob_info_json = ingestion_blob_info.to_json()
-        queue_client = queue_service.get_queue_client(queue=random_queue.object_name, message_encode_policy=TextBase64EncodePolicy())
-        queue_client.send_message(content=ingestion_blob_info_json, timeout=self._SERVICE_CLIENT_TIMEOUT_SECONDS)
+        with QueueServiceClient(random_queue.account_uri, proxies=self._proxy_dict) as queue_service:
```

Comment:
Should we not reuse those?

Reply:
I can make a cache for them, but just to be clear - they weren't re-used before, they were only not closed.

Reply:
So every time we enqueue a message we create a new blob client?
Seems like something that needs to be encapsulated by the resource manager, no?
Separate PR for sure though

Reply:
I'll open an issue